common/models/acl.js

```diff
@@ -506,12 +506,19 @@ module.exports = function(ACL) {
   };
 
   ACL.resolveRelatedModels = function() {
-    if (!this.roleModel) {
-      var reg = this.registry;
-      this.roleModel = reg.getModelByType(loopback.Role);
-      this.roleMappingModel = reg.getModelByType(loopback.RoleMapping);
-      this.userModel = reg.getModelByType(loopback.User);
-      this.applicationModel = reg.getModelByType(loopback.Application);
+    var self = this;
+    if (!self.roleModel) {
+      var reg = self.registry;
+      var builtInModels = {
+        roleModel: reg.getModel('Role'),
+        roleMappingModel: reg.getModel('RoleMapping'),
+        userModel: reg.getModel('User'),
+        applicationModel: reg.getModel('Application'),
+      };
+
+      for (var m in builtInModels) {
+        self[m] = reg.getModelByType(builtInModels[m]);
```

Comment:
my bad. I always read it as 
```
if (type === 'string') {
    return this.getModel(modelType);
}
```
and also mixed my assumptions with the previous implementation that used 
```
reg.getModelByType(loopback.Role);
```
which would return `loopback.Role` instead of `reg.getModel('Role')` model from because `loopback.Role` is neither of type `string` nor is there any model in `reg` that would have `loopback.Role` for it's prototype.